bot/decorators.py

```diff
@@ -105,15 +98,15 @@ def wrap(func):
     return wrap
 
 
-def redirect_output(destination_channel: int, bypass_roles: typing.Container[int] = None):
-    """
-    Changes the channel in the context of the command to redirect the output
-    to a certain channel, unless the author has a role to bypass redirection
+def redirect_output(destination_channel: int, bypass_roles: Container[int] = None) -> Callable:
     """
+    Changes the channel in the context of the command to redirect the output to a certain channel.
 
-    def wrap(func):
+    Redirect is bypassed if the author has a role to bypass redirection.
+    """
+    def wrap(func: Callable) -> Callable:
         @wraps(func)
-        async def inner(self, ctx, *args, **kwargs):
+        async def inner(self: Callable, ctx: Context, *args, **kwargs) -> Callable:
```

Comment:
This `inner` returns `None`.

Interestingly, I did not think to capture the return value of `func` and return it, which would probably have been a better practice (then, again, we should only apply this decorator to commands, which we do sometimes allow to return something, but I don't think we ever use the return value).

```suggestion
        async def inner(self: Callable, ctx: Context, *args, **kwargs) -> None:
```